pysatochip/CardConnector.py

```diff
@@ -1996,7 +2054,17 @@ def card_verify_authenticity(self):
         else:
             is_valid_serial_number= False
             txt_error= f"Certificate subject {subject} does not match the card serial number {self.UID_SHA1}!"
+
+        ## check the certificate chain from root CA to device
+        is_valid_chain, device_pubkey, txt_ca, txt_subca, txt_device, txt_error= validator.validate_certificate_chain(cert_pem, self.card_type)
+        if not is_valid_chain:
+            return is_valid_serial_number, txt_ca, txt_subca, txt_device, txt_error
```

Comment:
return is_valid_serial_number, txt_ca, txt_subca, txt_device, txt_error => return False

Reply:
Ah yes, in terms of this change, I'm not clear on what you are communicating with => but I can make a fix, on that note, basically you will noticed that the order of the checking was redone a bit.

Basically the rationale for that is that it's worth reporting things like the certificate matching the serial number, even if the certificate itself can't be fully verified for some other reason. (So basically giving the ability to still have it be marked as untrusted, but add some extra context for folk debugging their own certs)